src/types.rs

```diff
@@ -848,6 +848,71 @@ impl OwnedReceiveName {
     pub fn as_ref(&self) -> ReceiveName { ReceiveName(self.0.as_str()) }
 }
 
+/// An entrypoint name (borrowed version). Expected format:
+/// "<func_name>" where
+#[derive(Eq, PartialEq, Debug, Clone, Copy, Hash)]
+pub struct EntrypointName<'a>(pub(crate) &'a str);
+
+impl<'a> EntrypointName<'a> {
+    /// Size of the name in bytes.
+    pub fn size(&self) -> u32 { self.0.as_bytes().len() as u32 }
```

Comment:
This should be equivalent. Did you add the `as_bytes()` to be more explicit?
```suggestion
    pub fn size(&self) -> u32 { self.0.len() as u32 }
```

Reply:
Yes, indeed (both for they are equivalent, and I added it to be explicit). length of strings is always ambiguous, so I thought it was better to be explicit.

Reply:
Right, then I think we should do the same in
```
    if name.len() > constants::MAX_FUNC_NAME_SIZE {
```